tests/datasource/fluent/_fake_cloud_api.py

```diff
@@ -384,7 +386,7 @@ def post_datasources_cb(
         )
 
 
-def put_datasource_cb(request: PreparedRequest) -> CallbackResult:
+def put_datasource_cb(request: PreparedRequest) -> CallbackResult:  # noqa: C901
```

Comment:
nit: You can avoid the noqa by putting the new functionality into a helper function.